lms/djangoapps/mobile_api/users/tests.py

```diff
@@ -275,6 +275,35 @@ def test_discussion_url(self):
         self.assertIn('/api/discussion/v1/courses/{}'.format(self.course.id), response_discussion_url)
 
 
+    def test_org_query(self):
+        self.login()
+
+        # Create list of courses with various organizations
+        courses = [
+            CourseFactory.create(org='edX', mobile_available=True),
+            CourseFactory.create(org='edX', mobile_available=True),
+            CourseFactory.create(org='edX', mobile_available=True, visible_to_staff_only=True),
+            CourseFactory.create(org='Proversity.org', mobile_available=True),
+            CourseFactory.create(org='MITx', mobile_available=True),
+            CourseFactory.create(org='HarvardX', mobile_available=True),
+        ]
+
+        # Enroll in all the courses
+        for course in courses:
+            self.enroll(course.id)
+
+        response = self.api_response(qargs={'org':'edX'})
+
+        # Verify only edX courses are returned
+        self.assertEqual(len(response.data), 3)
+        for course_index in range(3):
+            result = response.data[course_index]['course']
```

Comment:
Since you've verified that there are objects in `response.data`, better to loop directly over the collection: 

    for entry in response.data:
        course = entry['course']

That way if the data set gets modified, that `range(3)` can't hide any errors.